imap_processing/tests/hit/test_hit_l1a.py

```diff
@@ -1,99 +1,201 @@
+def datasets(packet_filepath):
+    """Create datasets from packet file"""
+    packet_definition = (
+        imap_module_directory / "hit/packet_definitions/" "hit_packet_definitions.xml"
+    )
+    datasets_by_apid = packet_file_to_datasets(
+        packet_file=packet_filepath,
+        xtce_packet_definition=packet_definition,
+    )
+    return datasets_by_apid
 
 
-def test_create_datasets(unpacked_packets):
-    """Create xarray datasets
-
-    Creates a xarray dataset for each apid and stores
-    them in a dictionary
-
-    Parameters
-    ----------
-    unpacked_packets : list[space_packet_parser.parser.Packet]
-        A sorted list of decommutated packets
-    """
-    grouped_data = hit_l1a.group_data(unpacked_packets)
-
+@pytest.fixture(scope="module")
+def attribute_manager():
+    """Create the attribute manager"""
     attr_mgr = ImapCdfAttributes()
     attr_mgr.add_instrument_global_attrs(instrument="hit")
     attr_mgr.add_instrument_variable_attrs(instrument="hit", level="l1a")
     attr_mgr.add_global_attribute("Data_version", "001")
+    return attr_mgr
 
-    datasets_by_apid = hit_l1a.create_datasets(grouped_data, attr_mgr)
-    assert len(datasets_by_apid.keys()) == 1
-    assert isinstance(datasets_by_apid[hit_l1a.HitAPID.HIT_HSKP], xr.Dataset)
 
+@pytest.fixture(scope="module")
+def housekeeping_dataset(datasets):
+    """Get the housekeeping dataset"""
+    return datasets[HitAPID.HIT_HSKP]
 
-def test_hit_l1a(packet_filepath):
-    """Create L1A datasets
 
-    Creates xarray datasets from a packet.
+def test_hit_l1a(packet_filepath):
+    """Create L1A datasets from a packet file.
 
     Parameters
     ----------
     packet_filepath : str
         Path to ccsds file
     """
-    datasets = hit_l1a.hit_l1a(packet_filepath, "001")
-    assert len(datasets) == 1
-    assert isinstance(datasets[0], xr.Dataset)
-    assert datasets[0].attrs["Logical_source"] == "imap_hit_l1a_hk"
-
-
-def test_total_datasets(unpacked_packets):
-    """Test if total number of datasets is correct"""
-    # assert len(unpacked_packets) == total_datasets
-
-
-def test_dataset_dims_length(unpacked_packets):
-    """Test if the time dimension length in the dataset is correct"""
-    # grouped_data = hit_l1a.group_data(unpacked_packets)
-    # assert grouped_data["hit_l1a.HitAPID.HIT_HSKP"].dims["epoch"] == num_packet_times
+    processed_datasets = hit_l1a(packet_filepath, "001")
+    # TODO: update assertions after science data processing is completed
+    assert isinstance(processed_datasets, list)
+    assert len(processed_datasets) == 1
+    assert isinstance(processed_datasets[0], xr.Dataset)
+    assert processed_datasets[0].attrs["Logical_source"] == "imap_hit_l1a_hk"
+
+
+def test_concatenate_leak_variables(housekeeping_dataset):
+    """Test concatenation of leak_i variables"""
+
+    # Call the function
+    updated_dataset = concatenate_leak_variables(housekeeping_dataset)
```

Comment:
I think it'd be good to check that the values are correct. Maybe before you call this function, remove everything but the first couple packets for housekeeping and hardcode arrays for all the Leak values to check against

Reply:
Added!